packages/storybook/src/generators/configuration/util-functions.ts

```diff
@@ -325,34 +328,85 @@ export function createRootStorybookDir(
   }
 }
 
-export function createProjectStorybookDir(
+export function createRootStorybookDirForRootProjectInNestedWorkspace(
   tree: Tree,
   projectName: string,
   uiFramework: StorybookConfigureSchema['uiFramework'],
   js: boolean,
   tsConfiguration: boolean,
+  root: string,
+  projectType: string,
   isNextJs?: boolean,
   usesSwc?: boolean
 ) {
-  // Check if root main file is .ts or .js
-  if (tree.exists('.storybook/main.ts')) {
-    logger.info(
-      `The root Storybook configuration is in TypeScript, 
-      so Nx will generate TypeScript Storybook configuration files 
-      in this project's .storybook folder as well.`
+  const rootConfigExists =
+    tree.exists('.storybook/main.root.js') ||
+    tree.exists('.storybook/main.root.ts');
+  const rootProjectConfigExists =
+    tree.exists('.storybook/main.js') || tree.exists('.storybook/main.ts');
+
+  if (!rootConfigExists) {
+    createRootStorybookDir(tree, js, tsConfiguration);
+  }
+
+  if (rootConfigExists && rootProjectConfigExists) {
+    logger.warn(
+      `Storybook configuration files already exist for ${projectName}!`
     );
-    tsConfiguration = true;
-  } else {
-    if (tree.exists('.storybook/main.js')) {
-      logger.info(
-        `The root Storybook configuration is in JavaScript, 
-        so Nx will generate JavaScript Storybook configuration files 
-        in this project's .storybook folder as well.`
-      );
-      tsConfiguration = false;
-    }
+    return;
+  }
+  if (rootConfigExists && !rootProjectConfigExists) {
+    logger.warn(
+      `Root Storybook configuration files already exist. Will only generate project-specific configuration.`
```

Comment:
I think this reads better: `Only the project-specific configuration file will be generated.`